content/docs/instrumenting/pushing.md

```diff
@@ -21,3 +21,68 @@ For use from Java see the
 class.
 
 For use from Go see the [Push](http://godoc.org/github.com/prometheus/client_golang/prometheus#Push) and [PushAdd](http://godoc.org/github.com/prometheus/client_golang/prometheus#PushAdd) functions.
+
+## Java Batch Job Example
+
+This example illustrates how to instrument a batch job and alert on it not having succeeded recently.
+
+If using Maven, add the following to `pom.xml`:
+
+```
+        <dependency>
+            <groupId>io.prometheus</groupId>
+            <artifactId>simpleclient</artifactId>
+            <version>0.0.6</version>
+        </dependency>
+        <dependency>
+            <groupId>io.prometheus</groupId>
+            <artifactId>simpleclient_pushgateway</artifactId>
+            <version>0.0.6</version>
+        </dependency>
+```
+
+
+Instrument to track duration and last success:
+
+```java
+import io.prometheus.client.CollectorRegistry;
+import io.prometheus.client.Gauge;
+import io.prometheus.client.exporter.PushGateway;
+
+void executeBatchJob() throws Exception {
+ CollectorRegistry registry = new CollectorRegistry();
+ Gauge duration = Gauge.build()
+     .name("my_batch_job_duration_seconds")
+     .help("Duration of my batch job in seconds.")
+     .register(registry);
+ Gauge.Timer durationTimer = duration.startTimer();
+ try {
+   // Your code here.
+
+   // This is only added to the registry after success,
+   // so that a previous success in the Pushgateway is not overwritten on failure.
+   Gauge lastSuccess = Gauge.build()
+       .name("my_batch_job_last_success")
+       .help("Last time my batch job succeeded, in unixtime.")
+       .register(registry);
+   lastSuccess.setToCurrentTime();
+ } finally {
+   durationTimer.setDuration();
+   PushGateway pg = new PushGateway("127.0.0.1:9091");
+   pg.pushAdd(registry, "my_batch_job", "my_batch_job");
+ }
+}
+```
+
+Setup a Pushgateway, update the host and port in the above code if needed.
+
+Setup an alert to fire if the job has not run recently. Add the following to
+the rules of a Prometheus server that's scraping the Pushgateway:
+
+```
+ALERT MyBatchJobNotCompleted
+  IF min(time() - my_batch_job_last_success{job="my_batch_job"}) > 60 * 60
```

Comment:
I vote for `my_batch_job_last_success_timestamp_seconds`. Zero ambiguity about what the metric contains then.